src/applications/common/schemaform/fields/ArrayField.jsx

```diff
@@ -136,9 +152,9 @@ export default class ArrayField extends React.Component {
           ? false
           : val;
       });
-      const editingState = this.props.uiSchema['ui:options'].reviewMode;
+      const editingState = this.props.uiSchema['ui:options'].reviewMode ? true : 'adding';
       const newState = _.assign(this.state, {
-        editing: newEditing.concat(!!editingState)
+        editing: newEditing.concat(editingState)
```

Comment:
I think we might benefit from tweaking this state name and possible values with these changes. `true`, `false`, and `'adding'` seems confusing. Maybe we should call it `itemMode` and use `viewing`, `editing`, and `adding`?

Reply:
Thanks @jbalboni, that is much more expressive, updated.